File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0];
        int max_profit = 0;
        for(int i=0;i<prices.length;i++){
            max_profit = Math.max(max_profit,prices[i]-min);
            min = Math.min(min,prices[i]);
        }
        return max_profit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func findMin(a, b int) int {
    if(a < b) {
        return a;
    }
    return b;
}

func findMax(a, b int) int {
    if(a > b) {
        return a;
    }
    return b;
}


func maxProfit(prices []int) int {

    maxProfit := 0;
    minBuy := int(1e9);

    for _, price := range prices {
        minBuy = findMin(minBuy, price);
        maxProfit = findMax(maxProfit, price - minBuy);
    }
    return maxProfit;


    
}